### DISCUSSION TITLE
Directly visiting the root path returns a 500 error

### DISCUSSION THREAD

**Discussion by alex-r89** [apollographql/apollo-server#5479] [Q&A]
Hi there,

Before I begin, I just wanted to clarify that my apollo server is actually working fine, but I have a query which is confusing me and I was wondering if anyone could help:

I have a front end which runs on `https://mydomain.net` and a graphql apollo express server which runs on `https://api.mydomain.net`. As mentioned above, this (the apollo server) works fine **when I visit my domain url**. However, If i navigate in the browser directly to `https://api.mydomain.net` this returns a 500 error.

I do have an NGinx reverse proxy set up, but this is with the following which, afaik, should not result in a 500 error:
```
    location / {
        proxy_pass http://localhost:4000;
    }
```

Although this isn't causing any issues, this does actually result in my load balancer reporting that my apollo server is down (because pinging the root path via http/s returns `500`).

Could anyone clarify why this is and a potential fix? Even if this just returned a `204` that would be better I feel?

Thanks

**Comment by glasser** [comment]
Which [package](https://www.apollographql.com/docs/apollo-server/integrations/middleware/) are you using (apollo-server, apollo-server-express, etc)? What version of Apollo Server are you using?

**Reply by alex-r89** [reply]
Im using` apollo-server-express` version `2.23.0`

**Reply by glasser** [reply]
We improved handling of [bad request errors](https://www.apollographql.com/docs/apollo-server/migration/#bad-request-errors-more-consistently-return-4xx) in Apollo Server 3.  You didn't share the actual provided error but it's probably trying to tell you that it's trying to interpret your GET request as a GraphQL request but there's no query provided.

In addition to improving error handling (by using 4xx errors instead of 5xx in more cases where is makes sense), AS3 gives your server a nice landing page by default instead of trying to parse GET requests from a browser (with `accept: text/html`) as GraphQL requests.

**Reply by alex-r89** [reply]
Hi,

Thanks for the info. I don't see any error other than the 500 error in my console:

`api.mydomain.net/:1 GET https://api.mydomain.net/ 500 (Internal Server Error)`

This is after updating to Apollo Server 3. I'm honestly not sure if this is down to my apollo server or my Nginx reverse proxy now.

Saying that, I'm debating whether or not this is a bad thing; On the one hand its annoying that my load balancer is saying the servers are offline because the HTTP ping reports 500 (works fine if i do a TCP ping though). However on the other hand, I see random requests to my api endpoint at random paths, like `/php`, `/admin` and `/solr`  which return a 500, so I assume these random scanners just move on rather than reporting back any success response because I never see any follow up requests, which could indirectly be lessening the risk of an attacker putting more effort into my api.


Thanks for the help so far, really appreciate it.

**Reply by glasser** [reply]
Does your app log anything?

You can also use the built-in health check path for your load balancer: `/.well-known/apollo/server-health`
